pkg/controller/machine/controller.go

```diff
@@ -380,7 +380,7 @@ func (r *ReconcileMachine) Reconcile(ctx context.Context, request reconcile.Requ
 		if err := r.updateStatus(ctx, m, phaseProvisioning, nil, originalConditions); err != nil {
 			return reconcile.Result{}, err
 		}
-		return reconcile.Result{Requeue: true}, nil
+		return reconcile.Result{}, nil
```

Comment:
I think this needs a code warning to future spelunkers
```suggestion
		// DO NOT requeue here as the status update will trigger an event to cause this object to re-reconcile once the informer observes the update.
		// If we requeue here, we may see stale objects reconciled which can then lead to leaked instances.
		return reconcile.Result{}, nil
```

Reply:
Hmm, we already have a similar comment on the finalizer, https://github.com/openshift/machine-api-operator/blob/ca97d71a5be84f5c11f5ff75b9e5a0aa5c805aa0/pkg/controller/machine/controller.go#L203, that clearly didn't help us avoid this one 🤔

Reply:
...and now I understand why this recommendation (to return after every update to the API object) exists. I wouldn't have understood from reading this comment either fwiw.

I stand by: it's a general documentaiton and education issue.

Reply:
I'll yield